### PR TITLE
chore: update cdk api based on api review

### PR DESCRIPTION
**Pull request by alharris-at**
#### Description of changes
Update per api review:
* Rename `authRole`/`unauthRole` to `authenticatedUserRole`/`unauthenticatedUserRole`.
* Rename `TransformParameters` to `GraphqlBehavior`.
* Normalize capitalization of `GraphQL` to `Graphql` in comments, to align w/ AppSync L2 construct naming.
* Rename package from `@aws-amplify/graphql-api-construct-alpha` to `@aws-amplify/graphql-construct-alpha`.
* Rename `referencedFunctions` to `functionNameMap`.
* Removed `IdentityPool`/`IdentityPoolId` types, and just accept an `identityPoolId: string` in the `IAMAuthorizationConfig`.
* Removed `ConfigWithModelOverride` to reduce output space.
* Flipped output resources back from lists to maps, keyed by `logicalId`.
* Fixed types on Optimistic/Automerge Conflict Resolution Strategies (had the handlerTypes flipped).

And one style change in e2e tests file was to flip double quotes to single.

##### CDK / CloudFormation Parameters Changed
N/A

#### Issue #, if available
N/A

#### Description of how you validated changes
Unit Tests Pass.

#### Checklist
- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by phani-srikar — APPROVED** [review]

**Review by dpilch — APPROVED** [review]

File: packages/amplify-graphql-api-construct/API.md [hunk 1]
```diff
@@ -127,18 +124,26 @@ export type FunctionSlotBase = {
 export type FunctionSlotOverride = Partial<Pick<AppsyncFunctionProps, 'name' | 'description' | 'dataSource' | 'requestMappingTemplate' | 'responseMappingTemplate' | 'code' | 'runtime'>>;
 
 // @public
-export type IAMAuthorizationConfig = {
-    identityPool?: IdentityPool;
-    authRole?: IRole;
-    unauthRole?: IRole;
-    adminRoles?: IRole[];
+export type GraphqlBehavior = {
```
Comment (edwardfoyle) [inline packages/amplify-graphql-api-construct/API.md:127]: This isn't really changing GraphQL behavior. Maybe `SchemaTranslationBehavior`?

File: packages/amplify-graphql-api-construct/API.md [hunk 2]
```diff
@@ -36,27 +36,27 @@ export type AmplifyGraphqlApiProps = {
     schema: AmplifyGraphqlApiSchema;
     apiName?: string;
     authorizationConfig: AuthorizationConfig;
-    referencedFunctions?: Record<string, IFunction>;
+    functionNameMap?: Record<string, IFunction>;
     conflictResolution?: ConflictResolution;
     stackMappings?: Record<string, string>;
```
Comment (edwardfoyle) [inline packages/amplify-graphql-api-construct/API.md:41]: maybe this should be `stackNameMap` to be consistent with `functionNameMap`?

**Review by edwardfoyle — COMMENTED** [review]
couple nits, but this is fine for an alpha

File: packages/amplify-graphql-api-construct/API.md (hunk 2 above)
Comment (alharris-at) [reply packages/amplify-graphql-api-construct/API.md:41]: Discussed, and leaving this as `stackMappings` today to reflect what this is in the Amplify CLI realm. https://docs.amplify.aws/cli/graphql/override/#place-appsync-resolvers-in-custom-named-stacks

**Pull request merged**